lib/app/migrate.ts

```diff
@@ -0,0 +1,584 @@
+  derivedAccountId: number,
+  options: ArgumentsCamelCase<MigrateAppOptions>
+) {
+  SpinniesManager.init();
+
+  const { appIdToMigrate, projectName, projectDest } =
+    await handleMigrationSetup(derivedAccountId, options);
+
+  if (!appIdToMigrate) {
+    return;
+  }
+
+  const { migrationId, uidMap } = await handleMigrationProcess(
+    derivedAccountId,
+    appIdToMigrate
+  );
+
+  const buildId = await finalizeMigration(
+    derivedAccountId,
+    migrationId,
+    uidMap,
+    projectName,
+    options.platformVersion
+  );
+
+  if (!buildId) {
+    throw new Error('Migration Failed');
+  }
+
+  await downloadProjectFiles(
+    derivedAccountId,
+    projectName,
+    buildId,
+    projectDest
+  );
+}
+
+export function logInvalidAccountError(i18nKey: string) {
+  uiLine();
+  logger.error(i18n(`${i18nKey}.errors.invalidAccountTypeTitle`));
+  logger.log(
+    i18n(`${i18nKey}.errors.invalidAccountTypeDescription`, {
+      useCommand: uiCommandReference('hs accounts use'),
+      authCommand: uiCommandReference('hs auth'),
+    })
+  );
+  uiLine();
+}
+
+export async function migrateApp2023_2(
+  derivedAccountId: number,
+  options: ArgumentsCamelCase<MigrateAppOptions>,
+  accountConfig: CLIAccount
+) {
+  const i18nKey = 'commands.project.subcommands.migrateApp';
+  const accountName = uiAccountDescription(derivedAccountId);
+
+  if (!isAppDeveloperAccount(accountConfig)) {
+    logInvalidAccountError(i18nKey);
+    process.exit(EXIT_CODES.SUCCESS);
+  }
+
+  const { appId } =
+    'appId' in options
+      ? options
+      : await selectPublicAppPrompt({
+          accountId: derivedAccountId,
+          accountName,
+          isMigratingApp: true,
+        });
+
+  try {
+    const { data: selectedApp } = await fetchPublicAppMetadata(
+      appId,
+      derivedAccountId
+    );
+    // preventProjectMigrations returns true if we have not added app to allowlist config.
+    // listingInfo will only exist for marketplace apps
+    const preventProjectMigrations = selectedApp.preventProjectMigrations;
+    const listingInfo = selectedApp.listingInfo;
+    if (preventProjectMigrations && listingInfo) {
+      logger.error(i18n(`${i18nKey}.errors.invalidApp`, { appId }));
+      process.exit(EXIT_CODES.ERROR);
+    }
+  } catch (error) {
+    logError(error, new ApiErrorContext({ accountId: derivedAccountId }));
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  const createProjectPromptResponse = await createProjectPrompt(options);
+  const { name: projectName, dest: projectDest } = createProjectPromptResponse;
+
+  const { projectExists } = await ensureProjectExists(
+    derivedAccountId,
+    projectName,
+    {
+      allowCreate: false,
+      noLogs: true,
+    }
+  );
+
+  if (projectExists) {
+    throw new Error(
+      i18n(`${i18nKey}.errors.projectAlreadyExists`, {
+        projectName,
+      })
+    );
+  }
+
+  await trackCommandMetadataUsage(
+    'migrate-app',
+    { status: 'STARTED' },
```

Comment:
Same note as above--I think the closest allowed property would be `{ step: 'STARTED' }`.